### DISCUSSION TITLE
Menu - Looping children in .cshtml / Razor

### DISCUSSION THREAD

**Discussion by LMSSonos** [OrchardCMS/OrchardCore#9631] [Q&A] [answered]
I have a razor template `Menu.cshtml` with following code. I know that the items should then be rendered in the `MenuItem-**.cshtml` templates, but for debugging i wanted to output everything here. When I serialize the childLevel1 I get the data I need but I did not succeed to access the properties of the ContentMenuItemPart there. Any ideas?

```
@using OrchardCore.DisplayManagement
<ul>
@foreach (var item in Model.Items)
{
      <li>Parent: @item.ContentItem.Content.ContentMenuItemPart.Name</li>
      @foreach (var childLevel1 in item.ContentItem.Content.MenuItemsListPart.MenuItems)
      {
             <li>Child: @childLevel1.?????</li>
      }
}
</ul>
```

**Comment by ns8482e** [comment, marked as answer]
I believe `Model` and `Model.Items` are shapes. If you need to work with data objects of `Menu`, try using `Model.ContentItem.Content` which is json

**Reply by LMSSonos** [reply]
Thanks a lot for your help, this was exactly the issue, now works as expected!!
